src/features/pdf/PdfView2.tsx

```diff
@@ -40,39 +46,98 @@ export const PDFView2 = () => {
   if (pdfLayoutName) {
     // Render all components directly if given a separate PDF layout
     return (
-      <PdfWrapping>
-        <PlainPage pageKey={pdfLayoutName} />
-      </PdfWrapping>
+      <DataLoaderStoreInit>
+        <PdfWrapping>
+          <PlainPage pageKey={pdfLayoutName} />
+        </PdfWrapping>
+      </DataLoaderStoreInit>
     );
   }
 
   return (
-    <PdfWrapping>
-      <div className={classes.instanceInfo}>
-        <InstanceInformation
-          elements={{
-            dateSent: true,
-            sender: true,
-            receiver: true,
-            referenceNumber: true,
-          }}
-        />
-      </div>
-
-      {order
-        ?.filter((pageKey) => !isHiddenPage(pageKey))
-        .filter((pageKey) => !pdfSettings?.excludedPages.includes(pageKey))
-        .map((pageKey) => (
-          <PdfForPage
-            key={pageKey}
-            pageKey={pageKey}
-            pdfSettings={pdfSettings}
+    <DataLoaderStoreInit>
+      <PdfWrapping>
+        <div className={classes.instanceInfo}>
+          <InstanceInformation
+            elements={{
+              dateSent: true,
+              sender: true,
+              receiver: true,
+              referenceNumber: true,
+            }}
           />
-        ))}
-    </PdfWrapping>
+        </div>
+        {order
+          ?.filter((pageKey) => !isHiddenPage(pageKey))
+          .filter((pageKey) => !pdfSettings?.excludedPages.includes(pageKey))
+          .map((pageKey) => (
+            <PdfForPage
+              key={pageKey}
+              pageKey={pageKey}
+              pdfSettings={pdfSettings}
+            />
+          ))}
+        <SubformSummaryComponent2 />
+      </PdfWrapping>
+    </DataLoaderStoreInit>
   );
 };
 
+export function DataLoaderStoreInit({ children }: PropsWithChildren) {
+  const [loading, setLoading] = React.useState(true);
+  const subforms = useNodeTraversal((t) => t.allNodes().filter((node) => node.isType('Subform')));
+
+  const handleWorkerCompletion = React.useCallback(() => {
+    setLoading(false);
+  }, []);
+
+  return (
+    <>
+      {subforms.map((child, idx) => (
+        <DataLoaderStoreInitWorker
+          key={idx}
+          node={child}
+          initComplete={handleWorkerCompletion}
+        />
+      ))}
+      {!loading && children}
+    </>
+  );
+}
```

Comment:
@bjosttveit Since that part is my code, I might as well answer. You're right, and also it doesn't matter in this particular case, but if you see any easy improvement I think we should modify as suggested :)